account/service/user.go

```diff
@@ -2,21 +2,65 @@ package service
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/fabric8-services/fabric8-auth/application/service"
 
 	"github.com/fabric8-services/fabric8-auth/account/repository"
+	account "github.com/fabric8-services/fabric8-auth/account/repository"
 	"github.com/fabric8-services/fabric8-auth/application/service/base"
 	servicecontext "github.com/fabric8-services/fabric8-auth/application/service/context"
 	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/token"
+	"github.com/satori/go.uuid"
 )
 
 // NewUserService creates a new service to manage users
-func NewUserService(context servicecontext.ServiceContext) *userServiceImpl {
-	return &userServiceImpl{base.NewBaseService(context)}
+func NewUserService(ctx servicecontext.ServiceContext) service.UserService {
+	return &userServiceImpl{
+		BaseService: base.NewBaseService(ctx),
+	}
 }
 
 // userServiceImpl implements the UserService to manage users
 type userServiceImpl struct {
 	base.BaseService
+	tokenManager token.Manager
+}
+
+// UserInfo gets user information given a context containing access_token
+func (s *userServiceImpl) UserInfo(ctx context.Context, identityID uuid.UUID) (*account.User, *account.Identity, error) {
+	var user *account.User
+	var identity *account.Identity
+	err := s.ExecuteInTransaction(func() error {
+		var err error
+		identity, err = s.Repositories().Identities().Load(ctx, identityID)
```

Comment:
We have a method called IdentityWithUser(..) . Could we use that here ?

Reply:
Let me see, indeed

Reply:
@sbose78 I applied your suggestion in https://github.com/fabric8-services/fabric8-auth/pull/593/commits/7b7fda4942d86110c04f46c53645acb42246b163